ecommerce/extensions/payment/views/paystack.py

```diff
@@ -0,0 +1,135 @@
+
+Applicator = get_class('offer.applicator', 'Applicator')
+Basket = get_model('basket', 'Basket')
+NoShippingRequired = get_class('shipping.methods', 'NoShippingRequired')
+OrderTotalCalculator = get_class('checkout.calculators', 'OrderTotalCalculator')
+
+
+class PaystackExecutionView(EdxOrderPlacementMixin, View):
+    @property
+    def payment_processor(self):
+        return Paystack(self.request.site)
+
+    @method_decorator(transaction.non_atomic_requests)
+    def dispatch(self, request, *args, **kwargs):
+        return super(PaystackExecutionView, self).dispatch(request, *args, **kwargs)
+
+    def get_basket(self, basket_id):
+        """
+        Returna basket form basket_id
+        """
+        if not basket_id:
+            return None
+
+        try:
+            basket_id = int(basket_id)
+            basket = Basket.objects.get(id=basket_id)
+            basket.strategy = strategy.Default()
+            Applicator().apply(basket, basket.owner, self.request)
+            return basket
+        except (ValueError, ObjectDoesNotExist):
+            return None
+
+    def call_handle_order_placement(self, basket, request):
+        """
+        Handle order placement for approved transactions.
+        """
+        order_number = basket.order_number
+
+        try:
+            shipping_method = NoShippingRequired()
+            shipping_charge = shipping_method.calculate(basket)
+            order_total = OrderTotalCalculator().calculate(basket, shipping_charge)
+
+            user = basket.owner
+            order = self.handle_order_placement(
+                order_number=order_number,
+                user=user,
+                basket=basket,
+                shipping_address=None,
+                shipping_method=shipping_method,
+                shipping_charge=shipping_charge,
+                billing_address=None,
+                order_total=order_total,
+                request=request
+            )
+            self.handle_post_order(order)
+
+        except:  # pylint: disable=bare-except
+            self.log_order_placement_exception(order_number, basket.id)
+
+    def get(self, request):
+        """
+        Handle an incoming user returned to us by Paystack after approving payment.
+        It redirects user to order receipt page after completing all payment flow
+        of a successfull Paystack transaction.
+        """
+        reference = request.GET.get('reference')
+        success, response = self.payment_processor.paystack_client.handler(
+            VERIFY_TRANSACTION_CODE, reference
+        )
+
+        if success:
+            data = response.get('data')
+            metadata = data.get('metadata')
+
+            order_number = metadata.get('order_number')
+            basket_id = metadata.get('basket_id')
+            transaction_id = data.get('id')
+            logger.info(
+                'Received AuthorizeNet payment notification for transaction [%s], associated with basket [%d].',
```

Comment:
```suggestion
                "Received Paystack payment notification for transaction [%s], associated with basket [%d].",
```

Reply:
Why double quotes?

Reply:
Why are you suggesting the use of double quotes here ?

Reply:
Because she was using double-quote in every text string.